game-core/src/main/java/games/strategy/triplea/delegate/battle/MustFightBattle.java

```diff
@@ -1086,28 +1088,38 @@ private void pushFightLoopOnStack(final boolean firstRun) {
   private void addFightStartSteps(final boolean firstRun, final List<IExecutable> steps) {
     final boolean offensiveAa = canFireOffensiveAa();
     final boolean defendingAa = canFireDefendingAa();
-    if (offensiveAa) {
-      steps.add(
-          new FireOffensiveAaGuns() {
-            private static final long serialVersionUID = 3802352588499530533L;
-
-            @Override
-            public void execute(final ExecutionStack stack, final IDelegateBridge bridge) {
-              fireOffensiveAaGuns();
-            }
-          });
+    final BattleStep offensiveAaStep = new OffensiveAaFire(this, this);
+    final BattleStep defensiveAaStep = new DefensiveAaFire(this, this);
+    if (offensiveAaStep.valid()) {
+      steps.add(offensiveAaStep);
     }
-    if (defendingAa) {
-      steps.add(
-          new FireDefensiveAaGuns() {
-            private static final long serialVersionUID = -1370090785540214199L;
+    // see Save Game Compatibility Note on getBattleExecutables
+    new IExecutable() {
+      private static final long serialVersionUID = 3802352588499530533L;
 
-            @Override
-            public void execute(final ExecutionStack stack, final IDelegateBridge bridge) {
-              fireDefensiveAaGuns();
-            }
-          });
+      @Override
+      public void execute(final ExecutionStack stack, final IDelegateBridge bridge) {
+        // if this gets deserialized, then forward the work to the new BattleStep
```

Comment:
Given we already have the detailed comment above about save-game compatibility, do we very much need the 'if this gets' comment? 

I suspect we could delete it. The 'new' BattleStep mention is also very notable as it makes sense when looking at this diff, but a future maintainer looking at the post-merged code will not necessarily know have context to see the difference between 'new' and 'old'.